spec/armor_payments/api/resource_spec.rb

```diff
@@ -57,13 +61,13 @@ module ArmorPayments
 
       describe "#get" do
         it "queries the host for a specific resource, with approprate headers" do
-          Timecop.freeze(2014, 2, 22, 12, 0, 0) do
-            resource.connection.should_receive(:get).with({
+          Timecop.freeze(time) do
+            expect(resource.connection).to receive(:get).with({
               path: '/wibble/123/resource/456',
               headers: {
-                "X_ARMORPAYMENTS_APIKEY"            => "my-api-key",
-                "X_ARMORPAYMENTS_REQUESTTIMESTAMP"  => "2014-02-22T17:00:00Z",
-                "X_ARMORPAYMENTS_SIGNATURE"         => "48886620cfebb95ffd9ee351f4f68d4f103a8f4bdc0e3301f7ee709ec2cf3c19588ae1b67aa8ee38305de802651fb10093cf1af40f467ac936185d551a58a844"
+                "x-armorpayments-apikey"            => "my-api-key",
+                "x-armorpayments-requesttimestamp"  => "2014-02-22T12:00:00Z",
+                "x-armorpayments-signature"         => Digest::SHA512.hexdigest("#{api_secret}:GET:/wibble/123/resource/456:#{Time.now.utc.iso8601}")
```

Comment:
:+1:

It's not pedantry if it breaks things!

I assumed that `Timecop.freeze` would ensure that `Time.now` was consistent during the run of the test, but I can definitely see why you would prefer a less dynamic timestamp. 

Quick question, are specs green on master on your machine? I was getting `"2014-02-22T18:00:00Z"` when it expected `"2014-02-22T17:00:00Z"` in the specs, or vice versa. I thought it had to do with daylight savings time, so I specified a UTC offset of `"+00:00"`.

And if that's okay, is it okay to use that `"+00:00"` timestamp as the particular timestamp for the request? Instead of `Time.now.utc.iso8601` in the test, I'd use the string that it generates when frozen at that particular time.

More questions about the "previous header" in a bit.